channeldb/kvdb/etcd/readwrite_tx.go

```diff
@@ -25,7 +35,51 @@ func newReadWriteTx(stm STM) *readWriteTx {
 // rooBucket is a helper function to return the always present
 // root bucket.
 func rootBucket(tx *readWriteTx) *readWriteBucket {
-	return newReadWriteBucket(tx, rootBucketID())
+	return newReadWriteBucket(tx, rootBucketID(), rootBucketID())
+}
+
+// lock adds a key value to the lock set.
+func (tx *readWriteTx) lock(key, val string) {
+	tx.stm.Lock(key)
+	if !tx.dirty {
+		tx.lset = append(tx.lset, KV{key, val})
```

Comment:
Should this also be a map? IIRC there was an issue with using a slice for the main read/write set re duplicates.

Reply:
yes, indeed! fixed